integration-tests/gradle/src/integrationTest/kotlin/org/jetbrains/dokka/it/gradle/BasicCachingIntegrationTest.kt

```diff
@@ -24,6 +24,25 @@ class BasicCachingIntegrationTest(override val versions: BuildVersions) : Abstra
         runAndAssertOutcome(TaskOutcome.FROM_CACHE)
     }
 
+    @Test
+    fun localDirectoryPointingToRoot() {
+        fun String.findAndReplace(oldValue: String, newValue: String): String {
+            assertTrue(oldValue in this, "Expected to replace '$oldValue'")
+            return replace(oldValue, newValue)
+        }
+        val projectKts = projectDir.resolve("build.gradle.kts")
+        val projectKtsText = projectKts.readText()
+            .findAndReplace("localDirectory.set(file(\"src/main\"))", "localDirectory.set(projectDir)")
+            .findAndReplace("integration-tests/gradle/projects/it-basic/src/main", "integration-tests/gradle/projects/it-basic")
+        projectKts.writeText(projectKtsText)
+
+        runAndAssertOutcome(TaskOutcome.SUCCESS)
+        projectDir.resolve("unrelated.txt").writeText("modified")
+        // despite projectDir is used as an input in localDirectory, changing its contents shouldn't invalidate the cache
+        runAndAssertOutcome(TaskOutcome.FROM_CACHE)
```

Comment:
Nice test!

What do you think about adding a similar test or an additional step to this test to verify that changing the value of `localDirectory` invalidates the task? 

I doubt it'll happen, but Gradle could optimize out unused `@Input` properties or change the default behaviour - I'd like it to be caught with a test

Something like that:

```kotlin
...
runAndAssertOutcome(TaskOutcome.FROM_CACHE)

projectKts.readText()
            .findAndReplace("localDirectory.set(projectDir)", "localDirectory.set(file(\"src\"))")

runAndAssertOutcome(TaskOutcome.SUCCESS)
```

Reply:
I thought about it, but changing `localDirectory` this way will also change source links in the resulting html, so the content assertions won't pass. And changing `remoteUrl` to compensate that will make this change useless because it would be unclear which of these two properties has invalidated the task

Reply:
> the content assertions won't pass

Oh, I didn't know it checked for the actual values of the links, I thought it only asserted the task result... 

It doesn't look like we need to assert the output in this case though, this particular test can simply check that the cache was invalidated

WDYT about extracting the `run` part from `runAndAssertOutcome` into a separate function (without content checking), and using it for checking cache invalidation? Let it generate the wrong links as long as it **re**-generates them

Reply:
```kotlin
    private fun runAndAssertOutcome(expectedOutcome: TaskOutcome) {
        runTask(expectedOutcome)
        File(projectDir, "build/dokka/html").assertHtmlOutputDir()
    }

    private fun runTask(expectedOutcome: TaskOutcome) {
        val result = createGradleRunner(
            "clean",
            "dokkaHtml",
            "-i",
            "-s",
            "-Dorg.gradle.caching.debug=true",
            "--build-cache"
        ).buildRelaxed()

        assertEquals(expectedOutcome, assertNotNull(result.task(":dokkaHtml")).outcome)
    }
```

and then

```kotlin
        val projectKtsWithUpdatedLocalDirectory = projectKts.readText()
            .findAndReplace("localDirectory.set(projectDir)", "localDirectory.set(file(\"src/main/kotlin\"))")
        projectKts.writeText(projectKtsWithUpdatedLocalDirectory)
        runTask(TaskOutcome.SUCCESS)
```

Reply:
ok, I've followed this approach and split `runAndAssertOutcome` function into two.

Reply:
Thanks!